src/server/pfs/server/driver.go

```diff
@@ -299,27 +296,30 @@ func (d *driver) listRepoInTransaction(ctx context.Context, txnCtx *txncontext.T
 		return nil, errors.Wrap(err, "could not list repos of all types")
 	}
 	var repos []*pfs.RepoInfo
-	if err := stream.ForEach[pfsdb.RepoPair](ctx, iter, func(repoPair pfsdb.RepoPair) error {
+	if err := stream.ForEach[pfsdb.RepoWithID](ctx, iter, func(repoWithID pfsdb.RepoWithID) error {
+		if _, ok := projectNames[repoWithID.RepoInfo.Repo.Project.GetName()]; !ok && len(projectNames) > 0 {
```

Comment:
We should probably support filtering repos based on projects via SQL. Maybe have a `ListRepoInProjects` helper function? Or have a more generic filtering.

Reply:
Maybe, yeah. We can do that later though